crates/btc-relay/src/lib.rs

```diff
@@ -482,6 +482,34 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    // TODO: wrap sla update via staked-relayers pallet
+    fn _store_block_header_and_update_sla(
+        relayer: T::AccountId,
+        raw_block_header: RawBlockHeader,
+    ) -> DispatchResult {
+        if let Err(err) = Self::_store_block_header(relayer.clone(), raw_block_header) {
+            if err == DispatchError::from(Error::<T>::DuplicateBlock) {
+                // only accept duplicate if it is the chain head
+                let this_header_hash = raw_block_header.hash();
+                let best_header_hash = Self::get_best_block();
+                ensure!(
+                    this_header_hash == best_header_hash,
+                    Error::<T>::DuplicateBlock
```

Comment:
This error is a little bit misleading. Maybe rename to convey that the problem is that this is not the head

Reply:
I'm not sure anything else would make sense, do you have a suggestion?